### ISSUE TITLE
Import statement not being generated

### ISSUE THREAD

**Issue by woodforda** [mapstruct/mapstruct#2352]
- [x ] Is this an issue (and hence not a question)? 

Hi,

I have encountered an issue where an import is not being generated correctly.  It is relatively simple to replicate.

1. Domain classes.
````java
package org.woody.mapstruct.bug.dto;
...
@Value
public class TheDto {
    @NonNull
    String id;
}

package org.woody.mapstruct.bug.model;
...
@Value
public class TheModel {
    @NonNull
    String id;
}

package org.woody.mapstruct.bug.model;
...
public class TheModels extends ArrayList<TheModel> {
    private static final long serialVersionUID = 1L;
}
````

2. The mappers
````java
package org.woody.mapstruct.bug.mapper;
...
@Mapper
public interface TheModelMapper  {
    TheDto convert(TheModel theModel);
}

package org.woody.mapstruct.bug.mapper;
...
@Mapper(uses = TheModelMapper.class)
public interface TheModelsMapper  {
    List<TheDto> convert(TheModels theModels);
}
```` 

This causes the following code to be generated for the TheModelsMapperImpl
````java
package org.woody.mapstruct.bug.mapper;

import java.util.ArrayList;
import java.util.List;
import javax.annotation.Generated;
import org.mapstruct.factory.Mappers;
import org.woody.mapstruct.bug.dto.TheDto;
import org.woody.mapstruct.bug.model.TheModels;

@Generated(
    value = "org.mapstruct.ap.MappingProcessor",
    date = "2021-02-03T21:41:48+0100",
    comments = "version: 1.4.2.Final, compiler: javac, environment: Java 1.8.0_251 (Oracle Corporation)"
)
public class TheModelsMapperImpl implements TheModelsMapper {

    private final TheModelMapper theModelMapper = Mappers.getMapper( TheModelMapper.class );

    @Override
    public List<TheDto> convert(TheModels theModels) {
        if ( theModels == null ) {
            return null;
        }

        List<TheDto> list = new ArrayList<TheDto>( theModels.size() );
        for ( TheModel theModel : theModels ) {  // <-- note here we reference TheModel but we have no import for it
            list.add( theModelMapper.convert( theModel ) );
        }

        return list;
    }
}
```` 

I've noticed this as I'm moving across to using the Spring conversion concept and as a result have had to break up some of my mappers.  The above mapper without the uses generates all correctly as it will generate the additional mapping method which will cause the import to be created.

Currently I just have a dummy mapping method to force the import to occur.

I have a sample project if necessary.

cheers,
Andrew

**Comment by filiphr** [comment]
We'll look into the error.

In the meantime you can also use `Mapper#import`

**Comment by woodforda** [comment]
Ok.  Have put the Mapper#import to use as suggested.  Thanks.

**Comment by filiphr** [comment]
Thanks for the detailed report @woodforda, it helped a lot in writing the test. 

This will be fixed in PR @#2353

**Comment by woodforda** [comment]
You're welcome and thank you for the speedy response!!
